### PR TITLE
Fix cleanup! and prepare! Deprecations with Rails 5

### PR DESCRIPTION
**Pull request by atomaka**
Rails 5 deprecates `cleanup!` and `prepare!` causing a large amount of log spam from Delayed::Job:

```
DEPRECATION WARNING: cleanup! is deprecated and will be removed from Rails 5.1 (use Rails.application.reloader.reload! instead of cleanup + prepare) (called from require at bin/rails:9)
DEPRECATION WARNING: prepare! is deprecated and will be removed from Rails 5.1 (use Rails.application.reloader.prepare! instead) (called from require at bin/rails:9)
```

The first commit fixes existing issues with rubocop. Any easy corrections were made. Those that required refactorings were disabled.

### CODE REVIEW

**Comment by thomasdarde** [comment]
Hi , any reason for this PR not to be merged ?

**Comment by simonknoll** [comment]
@thomasdarde because the build failed.

**Comment by thomasdarde** [comment]
@atomaka Hi andrew, do you think you can update your PR for the build to pass ? Or tell me and I'll create another one

**Comment by atomaka** [comment]
I'll take a look this evening.  The build is failing because of a dependency issue on JRuby builds and, unfortunately, I haven't used JRuby.  I briefly look at this five months ago when I submitted the PR, but didn't see an immediate fix.  If this will be considered for merge, though, I'll spend a bit more time to get it to pass the builds.

For those that are still struggling with this, you can monkey patch on load to save yourselves from the log headaches:  https://github.com/collectiveidea/delayed_job/issues/944#issuecomment-265391660

**Comment by thomasdarde** [comment]
Thanks @atomaka , maybe just rebasing the PR will solve the issue, I hope it's not to much work.

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@
 .rvmrc
 /coverage
 Gemfile.lock
+.ruby-version
```
Comment (albus522) [inline .gitignore]: Nit pick, can we not include this please

File: Gemfile [hunk 2]
```diff
@@ -8,7 +10,13 @@ end
 
 platforms :jruby do
   gem 'jruby-openssl'
-  gem 'activerecord-jdbcsqlite3-adapter'
+  if rails_version == 'edge' || rails_version.match(/5\.\d+\.\d+/)
+    gem 'activerecord-jdbcsqlite3-adapter',
+        git: 'https://github.com/jruby/activerecord-jdbc-adapter.git',
+        branch: 'rails-5'
```
Comment (albus522) [inline Gemfile]: Maybe add a comment here about when this can be taken out.

**Comment by albus522** [comment]
Sorry for the inactivity, things have been kind of crazy for me

File: .gitignore (hunk 1 above)
Comment (atomaka) [inline .gitignore]: This was just for my personal testing.  This isn't prepared. Seeing what it takes to pass build.

File: Gemfile (hunk 2 above)
Comment (atomaka) [inline Gemfile]: Yup. See first comment. Still in process of testing.  See comment I am about to type

**Comment by atomaka** [comment]
The issue here is that the activerecord-jdbc-adapter no longer works with Rails 5.  There's[ a long lived issue](https://github.com/jruby/activerecord-jdbc-adapter/issues/700) about it dating back to December 2015. It was closed for some reason, but the issue persists (as can bee seen in the comment history).  There is a branch that has Rails 5 support (that I am testing builds against now), but I don't know the support level on that branch.

It might be wiser to drop Rails 5 support with Jruby until they resolve the issue upstream.

**Comment by atomaka** [comment]
Only one more build failure. Rails 4.2.0 on Ruby 2.0 because of a Nokogiri issue.  I'm not sure how to resolve, especially since when I reproduce locally, the correct version of Nokogiri is installed.

```
|-127-> rm -f Gemfile.lock && rbenv local 2.0.0-p598 && RAILS_VERSION="~> 4.2.0" bundle install
[SNIP]
Installing nokogiri 1.6.8.1 with native extensions
[SNIP]
Bundle complete! 13 Gemfile dependencies, 51 gems now installed.
Use `bundle show [gemname]` to see where a bundled gem is installed.
```

Full version in gist: https://gist.github.com/atomaka/ba5f4909cb82e07347b6181ad2f5948f

**Comment by cseelus** [comment]
Still not in master to work with Rails 5.1? 🤔

**Comment by timhaines** [comment]
@albus522 What's required to get this released in a new version of the gem?

**Pull request merged**